service/frontend/nexus_handler.go

```diff
@@ -531,16 +542,43 @@ func (h *nexusHandler) nexusClientForActiveCluster(oc *operationContext, service
 	})
 }
 
-// convertNexusHandlerError converts any 5xx user handler error to a downstream error.
-func convertNexusHandlerError(t nexus.HandlerErrorType) nexus.HandlerErrorType {
-	switch t {
-	case nexus.HandlerErrorTypeDownstreamTimeout,
-		nexus.HandlerErrorTypeUnauthenticated,
-		nexus.HandlerErrorTypeUnauthorized,
-		nexus.HandlerErrorTypeBadRequest,
-		nexus.HandlerErrorTypeNotFound,
-		nexus.HandlerErrorTypeNotImplemented:
-		return t
-	}
-	return nexus.HandlerErrorTypeDownstreamError
+func handlerErrorFromClientError(err error) (string, error) {
+	var unexpectedRespErr *nexus.UnexpectedResponseError
+	if errors.As(err, &unexpectedRespErr) {
+		failure := unexpectedRespErr.Failure
+		if unexpectedRespErr.Failure == nil {
+			failure = &nexus.Failure{
+				Message: unexpectedRespErr.Error(),
+			}
+		}
+		handlerErr := &nexus.HandlerError{
+			Failure: failure,
+		}
+
+		switch unexpectedRespErr.Response.StatusCode {
+		case http.StatusBadRequest:
+			handlerErr.Type = nexus.HandlerErrorTypeBadRequest
+		case http.StatusUnauthorized:
+			handlerErr.Type = nexus.HandlerErrorTypeUnauthenticated
+		case http.StatusForbidden:
+			handlerErr.Type = nexus.HandlerErrorTypeUnauthorized
+		case http.StatusNotFound:
+			handlerErr.Type = nexus.HandlerErrorTypeNotFound
+		case http.StatusTooManyRequests:
+			handlerErr.Type = nexus.HandlerErrorTypeResourceExhausted
+		case http.StatusInternalServerError:
+			handlerErr.Type = nexus.HandlerErrorTypeInternal
+		case http.StatusNotImplemented:
+			handlerErr.Type = nexus.HandlerErrorTypeNotImplemented
+		case http.StatusServiceUnavailable:
+			handlerErr.Type = nexus.HandlerErrorTypeUnavailable
+		case nexus.StatusDownstreamTimeout:
+			handlerErr.Type = nexus.HandlerErrorTypeDownstreamTimeout
+		}
```

Comment:
I see we missed this before but we should have a default case that converts to internal error (both for the type and the failure message).

Reply:
And log when that happens.